contrib/src/templates/watch.rs

```diff
@@ -0,0 +1,41 @@
+extern crate notify;
+
+use self::notify::{watcher, DebouncedEvent, RecommendedWatcher, RecursiveMode, Watcher};
+
+use std::path::Path;
+use std::sync::mpsc::{channel, Receiver};
+use std::sync::Mutex;
+use std::time::Duration;
+
+pub struct TemplateWatcher {
```

Comment:
Do we really need this extra layer of indirection?

Reply:
By extra layer do you mean the `watch.rs` file, or the entire structure? The motivation for this was to keep the `notify` dependency relatively isolated in the code.

Reply:
These fields and/or behavior are small enough they could maybe move into the debug-only version of `mod context` in `fairing.rs`.

Reply:
Yeah, that's what I'm thinking as well. Play around with it a bit. If we don't need the indirection, and it doesn't make code significantly cleaner, let's get rid of it.